exporters/otlp/testing-internal/src/main/java/io/opentelemetry/exporter/otlp/testing/internal/FakeTelemetryUtil.java

```diff
@@ -69,23 +74,15 @@ public static SpanData generateFakeSpanData() {
         .setLinks(Collections.emptyList())
         .setTotalRecordedLinks(0)
         .setTotalRecordedEvents(0)
-        .setInstrumentationScopeInfo(
-            InstrumentationScopeInfo.builder("testLib")
-                .setVersion("1.0")
-                .setSchemaUrl("http://url")
-                .build())
+        .setInstrumentationScopeInfo(SCOPE_INFO)
         .build();
   }
 
   /** Generate a fake {@link LogRecordData}. */
   public static LogRecordData generateFakeLogRecordData() {
     return TestLogRecordData.builder()
         .setResource(Resource.getDefault())
-        .setInstrumentationScopeInfo(
-            InstrumentationScopeInfo.builder("testLib")
-                .setVersion("1.0")
-                .setSchemaUrl("http://url")
-                .build())
+        .setInstrumentationScopeInfo(SCOPE_INFO)
```

Comment:
This change seems random, but is actually oddly needed for a subtle reason. The OTLP exporter tests generate fake test data, serialize and send it to a mock server, where it is deserialized into generated proto classes and evaluated for equality. As you're probably aware, OTLP payloads consist of a series of enclosing envelopes, e.g. for logs ResourceLogs enclose ScopeLogs enclose LogRecords. When an OTLP exporter recieves a `List<LogRecordData>`, the records need to be grouped by resource and scope to match the OTLP payload structure. We do this using _identity hash maps_. Notably, the reuseable memory mode serialization iterates over identity the identity hash maps slightly differently than the immutable memory mode variant. This means that when a group of records is exported with scopes which are equal but not identical, the order of those ScopeLogs in the serialized payload is not the same for immutable memory mode and reuseable memory mode.

We can make the tests pass by having the fake log records use identical scopes, which sidesteps the difference in iteration order between immutable memory mode and reuseable memory mode.

Reply:
Thanks for the clarification....this is a clever workaround.